components/director/internal/domain/runtime/service.go

```diff
@@ -1,8 +1,165 @@
+	Create(item *RuntimeModel) error
+	Update(item *RuntimeModel) error
+	Delete(item *RuntimeModel) error
+}
+
+type RuntimeInput struct {
+	Name        string              `json:"name"`
+	Description *string             `json:"description"`
+	Labels      map[string][]string `json:"labels"`
+	Annotations map[string]string   `json:"annotations"`
+}
+
 type Service struct {
+	repo Repository
+}
+
+func NewService(repo Repository) *Service {
+	return &Service{repo: repo}
+}
+
+func (s *Service) List(ctx context.Context, filter []*labelfilter.LabelFilter, pageSize *int, cursor *string) (*RuntimePage, error) {
+	return s.repo.List(filter, pageSize, cursor)
 }
 
-func NewService() *Service {
-	return &Service{}
+func (s *Service) Get(ctx context.Context, id string) (*RuntimeModel, error) {
+	runtime, err := s.repo.GetByID(id)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while getting Runtime with ID")
+	}
+
+	return runtime, nil
+}
+
+func (s *Service) Create(ctx context.Context, in RuntimeInput) (string, error) {
+	id := uuid.New().String()
+	runtimeTenant, err := tenant.LoadFromContext(ctx)
+	if err != nil {
+		return "", errors.Wrapf(err, "while loading tenant from context")
+	}
+
+	r := NewRuntimeModel(id, in.Name, in.Description, runtimeTenant, in.Labels, in.Annotations)
+
+	err = s.repo.Create(r)
+	if err != nil {
+		return "", err
+	}
+
+	return id, nil
+}
+
+func (s *Service) Update(ctx context.Context, id string, in RuntimeInput) error {
+	r, err := s.Get(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	r.Name = in.Name
+	r.Description = in.Description
+	r.Labels = in.Labels
+	r.Annotations = in.Annotations
+
+	err = s.repo.Update(r)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *Service) Delete(ctx context.Context, id string) error {
+	r, err := s.Get(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	return s.repo.Delete(r)
+}
+
+func (s *Service) AddLabel(ctx context.Context, runtimeID string, key string, values []string) error {
+	r, err := s.Get(ctx, runtimeID)
+	if err != nil {
+		return err
+	}
+
+	if _, exists := r.Labels[key]; exists {
+		err = r.AppendLabelValues(key, values)
+		if err != nil {
+			return errors.Wrapf(err, "while appending label values")
+		}
+	} else {
+		err = r.AddLabel(key, values)
+		if err != nil {
+			return errors.Wrapf(err, "while adding new label %s", key)
+		}
+	}
+
+	err = s.repo.Update(r)
+	if err != nil {
+		return errors.Wrapf(err, "while updating Runtime")
+	}
+
+	return nil
+}
+
+func (s *Service) DeleteLabel(ctx context.Context, runtimeID string, key string, values []string) error {
+	r, err := s.Get(ctx, runtimeID)
+	if err != nil {
+		return err
+	}
+
+	err = r.DeleteLabel(key)
+	if err != nil {
+		return errors.Wrapf(err, "while deleting label with key %s", key)
+	}
+
+	return nil
+}
+
+func (s *Service) AddAnnotation(ctx context.Context, runtimeID string, key string, value string) error {
+	r, err := s.Get(ctx, runtimeID)
```

Comment:
short names are cool, but sometimes they are too short. Maybe instead of r, we can use `model` name, or `rtm`?